src/EditorFeatures/Core/Remote/SolutionChecksumUpdater.cs

```diff
@@ -141,9 +141,10 @@ private async Task SynchronizePrimaryWorkspaceAsync(CancellationToken cancellati
 
             using (Logger.LogBlock(FunctionId.SolutionChecksumUpdater_SynchronizePrimaryWorkspace, cancellationToken))
             {
+                var workspaceVersion = solution.WorkspaceVersion;
                 await client.TryInvokeAsync<IRemoteAssetSynchronizationService>(
                     solution,
-                    (service, solution, cancellationToken) => service.SynchronizePrimaryWorkspaceAsync(solution, cancellationToken),
+                    (service, solution, cancellationToken) => service.SynchronizePrimaryWorkspaceAsync(solution, workspaceVersion, cancellationToken),
```

Comment:
genuine API change.  When the ohst wants to update what OOP sees as the primary workspace, it passes along the workspace version so that OOP only moves forward.